code/core/system/proxy-velodyne16/src/velodyne16Decoder.cpp

```diff
@@ -234,6 +249,8 @@ void Velodyne16Decoder::sendPointCloud() {
 
 void Velodyne16Decoder::nextString(const string &payload) {
     if (payload.length() == 1206) {
+        TimeStamp newPacket;
+        m_packetArrivalTime = newPacket.toMicroseconds();
```

Comment:
This is the time stamp when you entered this method. Is there a more precise way to get the time stamp from Velodyne?